src/v/cluster/rm_stm.cc

```diff
@@ -1710,7 +1710,7 @@ ss::future<> rm_stm::apply(model::record_batch b) {
             apply_data(bid, last_offset);
         }
     } else if (hdr.type == model::record_batch_type::tx_checkpoint) {
-        apply_checkpoint(b);
+        apply_checkpoint(std::move(b));
```

Comment:
Let's cache it instead of applying right away. in this case the sync will have a chance to reconcile the data before putting it into action instead of applying and reconciling them after the fact. The latter is more error prone.